src/main/java/io/pinecone/configs/PineconeConfig.java

```diff
@@ -0,0 +1,63 @@
+package io.pinecone.configs;
+
+import io.grpc.ManagedChannel;
+import io.pinecone.exceptions.PineconeValidationException;
+
+public class PineconeConfig {
+
+    private String apiKey;
+    private String host;
+    private String usageContext;
+    private ManagedChannel customChannelBuilder;
+
+    public PineconeConfig(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public String getApiKey() {
+        return apiKey;
+    }
+
+    public void setApiKey(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public String getHost() {
+        return host;
+    }
+
+    public void setHost(String host) {
+        this.host = host;
+    }
+
+    public String getUsageContext() {
+        return usageContext;
+    }
+
+    public void setUsageContext(String usageContext) {
+        this.usageContext = usageContext;
+    }
+
+    public ManagedChannel getCustomChannelBuilder() {
+        return this.customChannelBuilder;
+    }
+
+    public void setCustomChannelBuilder(ManagedChannel customChannelBuilder) {
+        this.customChannelBuilder = customChannelBuilder;
+    }
+
+    public interface CustomChannelBuilder {
+        ManagedChannel buildChannel();
+    }
```

Comment:
I'm confused about what's going on here. Shouldn't the return type of `getCustomChannelBuilder` be a `CustomChannelBuilder` instance which implements `buildChannel`? Seems like the interface you defined here is not being used for anything, which makes the method names quite confusing.

Reply:
We used to offer this functionality in PineconeConnectionConfig before. It allows the user to pass in their own gRPC ManagedChannel. If they don't provide it, we will create one for them within the PineconeConnection class. Also the reason why we have two constructors in PineconeConnection class.

Reply:
Good point, I'll rename it.